flox-bash/lib/utils.sh

```diff
@@ -1438,10 +1438,12 @@ function identifyParentShell() {
 	# Only attempt a guess if we know our parent PID.
 	if [ -n "${FLOX_PARENT_PID:-}" ]; then
 		# First attempt to identify details of parent shell process.
-		if [ -L "/proc/$FLOX_PARENT_PID/exe" -a \
-			 -r "/proc/$FLOX_PARENT_PID/exe" ]; then
-			# Linux - use information from /proc.
-			parentShell="$($_readlink "/proc/$FLOX_PARENT_PID/exe")"
+		if [ -r "/proc/$FLOX_PARENT_PID/cmdline" ]; then
+			# Linux - use information from /proc. bash can be invoked as sh, so we use
+			# argv[0] rather than /proc/$FLOX_PARENT_PID/exe
+			# -d '' splits on null bytes
+			mapfile -d '' cmdline < <($_cat "/proc/$FLOX_PARENT_PID/cmdline")
```

Comment:
Why use `$_cat` at all?
```suggestion
			mapfile -d '' cmdline < "/proc/$FLOX_PARENT_PID/cmdline"
```

Or if you like the error messages returned by cat (I do) then you can do this:
```suggestion
			$_cat "/proc/$FLOX_PARENT_PID/cmdline" | mapfile -d '' cmdline
```

Reply:
Ah ... cancel that second option ... won't work because the mapfile occurs in a subshell. I make that mistake all the time.